db/resourceDBInterface.go

```diff
@@ -0,0 +1,121 @@
+package db
+
+import (
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"os"
+)
+
+type CResource struct {
+	ID    primitive.ObjectID `bson:"_id,omitempty" json:"_id,omitempty"`
+	Name  string             `bson:"name" json:"name"`
+	Type  string             `bson:"type" json:"type"`
+	Users []User             `bson:"users" json:"users"`
```

Comment:
Why the users field ? instead I think it would make more sense to mock up the reservations model and store those. 

Also a description of the resource is needed. "

Type" should maybe be something enum like. My idea would be to have something like this 
```
type CResourceType int64 
const (
    gpu CResourceType = 0
    cpu CResourceType = 1
)

type CResource struct {
    ...
    Type CResourceType ...
}
```

Reply:
Fine for now and definitely better than last version. Might make sense to add logic for dividing resources into nodes (e.g. how many nodes do types consist of, how are the nodes specced, etc.), but as long as we dont get further information this is fine.